modules/table/src/test/java/org/apache/ignite/internal/table/MessagingServiceTestUtils.java

```diff
@@ -22,39 +22,64 @@
 import static org.mockito.ArgumentMatchers.anyLong;
 import static org.mockito.Mockito.doAnswer;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
+import java.util.Iterator;
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import java.util.function.BiConsumer;
+import org.apache.ignite.internal.table.distributed.command.FinishTxCommand;
+import org.apache.ignite.internal.table.distributed.raft.PartitionListener;
 import org.apache.ignite.internal.tx.TxManager;
 import org.apache.ignite.internal.tx.message.TxFinishRequest;
 import org.apache.ignite.internal.tx.message.TxFinishResponse;
+import org.apache.ignite.network.ClusterService;
 import org.apache.ignite.network.MessagingService;
 import org.apache.ignite.network.NetworkAddress;
+import org.apache.ignite.raft.client.Command;
+import org.apache.ignite.raft.client.service.CommandClosure;
 import org.mockito.Mockito;
 
 /**
  * Test utils for mocking messaging service.
  */
 public class MessagingServiceTestUtils {
-
     /**
      * Prepares messaging service mock.
      *
      * @param txManager Transaction manager.
      * @return Messaging service mock.
      */
-    public static MessagingService mockMessagingService(TxManager txManager) {
+    public static MessagingService mockMessagingService(
+            ClusterService clusterService,
+            TxManager txManager,
+            List<PartitionListener> partitionListeners
+    ) {
         MessagingService messagingService = Mockito.mock(MessagingService.class, RETURNS_DEEP_STUBS);
 
+        Mockito.when(clusterService.messagingService()).thenReturn(messagingService);
+
         doAnswer(
                 invocationClose -> {
                     assert invocationClose.getArgument(1) instanceof TxFinishRequest;
 
                     TxFinishRequest txFinishRequest = invocationClose.getArgument(1);
 
+                    FinishTxCommand finishTxCommand = new FinishTxCommand(
```

Comment:
Please be consistent with the order of txState change and releasing locks you've introduced in org.apache.ignite.internal.table.distributed.raft.PartitionListener#handleFinishTxCommand

Reply:
Seems that's fixed.